### PR TITLE
Implement select input

### PR DESCRIPTION
**Pull request by AdmiralCoco**
Slotted input for multi-select
Allows a custom input to used if desired

### CODE REVIEW

**Comment by AdmiralCoco** [comment]
Hey @dbatiste @dlockhart, would you mind giving this a review? 

(I've added the inputs and updated the README with some usage information. Also not sure why tests are failing on CI as they are passing locally)

File: README.md [hunk 1]
```diff
@@ -86,3 +141,4 @@ npm test
 [bower-image]: https://badge.fury.io/bo/d2l-multi-select.svg
 [ci-url]: https://travis-ci.org/BrightspaceUI/multi-select
 [ci-image]: https://travis-ci.org/BrightspaceUI/multi-select.svg?branch=master
+[design.d2l-url]: http://design.d2l/components/tags/
```
Comment (dbatiste) [inline README.md]: Is there a plan to update the name in the design doc?

Comment (AdmiralCoco) [reply README.md]: @cheics-d2l, do you have any thoughts about this?

Comment (cheics-d2l) [reply README.md]: Technically, the design doc is only for tags, not the multi-select component as created.

Note the "Multi-Select Input Field" mentioned at the bottom of the page - I can spin up a quick page around the component. When I was originally tackling it - I got stymied by autocompletion - it blew my mind how many variants there are.

I can document the specific control and NOT the autocomplete behaviour relatively easy. When I do - I'll send out an update so we can update the link.

**Comment by cheics-d2l** [comment]
> Cool - Colin already had some comments regarding accessibility which I will address in future PRs.
> Thanks for all the feedback 🙂

I'm meeting with Carin today to get some feedback around other product's implementations of multi-select and the discoverability of the delete action. The guidelines will be documented on Design.D2L.

**Pull request merged**